tests/unit/plugins/modules/test_zendesk_ticket.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2024, Luis Valle <levalle232@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible_collections.community.general.plugins.modules import zendesk_ticket
+from ansible_collections.community.general.tests.unit.plugins.modules.utils import ModuleTestCase, set_module_args, AnsibleExitJson, AnsibleFailJson
+from httmock import HTTMock, response, urlmatch
```

Comment:
httmock is a library for mocking requests, but this module is not using requests. My guess is that's the reason why the HTTP request mocking fails.

Reply:
that would make sense, should i go ahead and convert over to the requests library? i wanted to stick to Ansible base libraries to reduce requirements

Reply:
My bad, sorry, I thought httmock was kind of an "universal" mock library for HTTP requests, not specific for the `requests` library.

Reply:
> that would make sense, should i go ahead and convert over to the requests library? i wanted to stick to Ansible base libraries to reduce requirements

Don't use requests if not absolutely necessary. Using the Ansible base libraries is always preferred over using third-party libraries such as requests.